src/borg/archiver.py

```diff
@@ -1392,23 +1392,38 @@ def _delete_repository(self, args, repository):
 
         if not args.cache_only:
             if args.forced == 0:  # without --force, we let the user see the archives list and confirm.
+                id = bin_to_hex(repository.id)
+                location = repository._location.canonical_path()
                 msg = []
+                n_archives = None
                 try:
                     manifest, key = Manifest.load(repository, Manifest.NO_OPERATION_CHECK)
+                    n_archives = len(manifest.archives)
+                    msg.append(f"You requested to completely DELETE the following repository "
+                               f"*including* {n_archives} archives it contains:")
                 except NoManifestError:
-                    msg.append("You requested to completely DELETE the repository *including* all archives it may "
-                               "contain.")
-                    msg.append("This repository seems to have no manifest, so we can't tell anything about its "
-                               "contents.")
-                else:
-                    if self.output_list:
-                        msg.append("You requested to completely DELETE the repository *including* all archives it "
-                                   "contains:")
-                        for archive_info in manifest.archives.list(sort_by=['ts']):
-                            msg.append(format_archive(archive_info))
+                    msg.append("You requested to completely DELETE the following repository "
+                               "*including* all archives it may contain:")
+
+                msg.append(DASHES)
+                msg.append(f"Repository ID: {id}")
+                msg.append(f"Location: {location}")
+
+                if self.output_list:
+                    msg.append("")
+                    msg.append("Archives:")
+
+                    if n_archives is not None:
+                        if manifest.archives:
```

Comment:
as you've determined `n_archives` already, you could also use it here, like `if n_archives > 0:`.

Reply:
I think this is not possible because we are checking 3 states here:

- If manifest is not available, we tell user that we were unable to check.
- If we have manifest and therefore `n_archives >= 0` we want the output with `Archives:`
- ...but in case there aren't any archives, display a friendly message indicating that the missing listing is not an application error -- there just aren't any archives to list.